.github/workflows/spell-checker.yml

```
@@ -1,25 +1,33 @@
-name: Fix typos nightly
+name: Spell Check
 
 on:
-  workflow_dispatch: {} # enable manual running of workflow.
+  pull_request:
   schedule:
-    - cron: "0 8 * * *"
+    - cron: "0 12 * * *"
+
+permissions:
+  contents: write
+  pull-requests: write
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
-      - uses: sobolevn/misspell-fixer-action@26173e641ba08255d0c9826b8a873f8b96f76932
+      - name: Run codespell
+        continue-on-error: true
+        run: |
+          sudo apt-get install codespell -y
+          codespell -w --skip="*.pulsar.go,*.pb.go,*.pb.gw.go,*.cosmos_orm.go,*.json,*.git,*.js,crypto/keys,fuzz,*.h,proto/tendermint,*.bin,go.sum,go.mod,.github" --ignore-words=.github/.codespellignore
```

lets remove some unecessary extensions here

```suggestion
          codespell -w --skip="*.pb.go,*.pb.gw.go,*.json,*.git,*.js,*.h,*.bin,go.sum,go.mod,.github" --ignore-words=.github/.codespellignore
```